lasagne/layers/special.py

```diff
@@ -648,9 +658,25 @@ def __init__(self, incoming, localization_network, downsample_factor=1,
                              "output shape: (batch_size, num_input_channels, "
                              "input_rows, input_columns)")
 
+        if type(precompute_grid) is not bool and precompute_grid != 'auto':
+            raise ValueError("precompute_grid must be a boolean or set to "
+                             "'auto'.")
```

Comment:
I think I wouldn't do an explicit type check. Sometimes you'll want `0` to be an accepted false-ish value. What about a structure as follows:
```python
can_precompute = all(s is not None for s in input_shp[2:])
if precompute_grid == 'auto':
    precompute_grid = can_precompute
elif precompute_grid and not can_precompute:
    raise ValueError(...)
self.precompute_grid = precompute_grid
```